pkg/controllers/dynakube/extension/secret.go

```diff
@@ -0,0 +1,83 @@
+package extension
+
+import (
+	"context"
+
+	"github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta3/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/dttoken"
+	k8ssecret "github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/secret"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eecTokenKey         = "eec-token"
+	eecTokenValuePrefix = "EEC dt0x01"
+	secretSuffix        = "-extensions-token"
+)
+
+func reconcileSecret(ctx context.Context, dk *dynakube.DynaKube, kubeClient client.Client, apiReader client.Reader) error {
+	log.Info("reconciling secret" + getSecretName(dk.Name))
+
+	query := k8ssecret.NewQuery(ctx, kubeClient, apiReader, log)
+
+	if !dk.PrometheusEnabled() {
+		err := query.Delete(getSecretName(dk.Name), dk.Namespace)
+		if err != nil {
+			return err
+		}
+
+		removeSecretCreatedCondition(dk.Conditions())
+
+		return nil
+	}
+
+	_, err := query.Get(client.ObjectKey{Name: getSecretName(dk.Name), Namespace: dk.Namespace})
+	if err != nil && !errors.IsNotFound(err) {
+		setSecretCreatedFailureCondition(dk.Conditions(), err)
+
+		return err
+	}
+
+	if errors.IsNotFound(err) {
+		log.Info("creating secret" + getSecretName(dk.Name))
```

Comment:
space is missing

Reply:
Fixed [here](https://github.com/Dynatrace/dynatrace-operator/pull/3485/commits/24ace1ad5534ddee99b63515d3e7b766683561bc)